services/opentsdb/service.go

```diff
@@ -297,20 +341,48 @@ func (s *Service) Addr() net.Addr {
 }
 
 // serve serves the handler from the listener.
-func (s *Service) serve() {
-	for {
-		// Wait for next connection.
-		conn, err := s.ln.Accept()
-		if opErr, ok := err.(*net.OpError); ok && !opErr.Temporary() {
-			s.Logger.Info("OpenTSDB TCP listener closed")
-			return
-		} else if err != nil {
-			s.Logger.Info("Error accepting OpenTSDB", zap.Error(err))
-			continue
+func (s *Service) serve(ctx context.Context) error {
+	conChan := make(chan net.Conn)
+	errChan := make(chan error)
+
+	// send accepted connections to our conChan so that we can multiplex over
+	// conChand and ctx.Done() in the next for loop.
+	//
+	// this go routine is cancelled once the s.ln listener is closed.
+	//
+	// FIXME: this logic needs to be looked at closely.
```

Comment:
> this logic needs to be looked at closely.

looked at closely, or tested comprehensively and they called good? 😄

Reply:
just looked at. not even read.  just look at the lines and curves that form the letters and words but without assigning any meaning to them.